### ISSUE TITLE
Regression: Type cheking of binary operations involving numpy scaler types fails with 1.1.336

### ISSUE THREAD

**Issue by jenshnielsen** [microsoft/pyright#6466]
**Describe the bug**
The following code is example of binary operations involving numpy types that used to typecheck but no longer does with 1.1.336
This does not seem limited to these types or operations, but they are given as an example.

**Code or Screenshots**

```python
import numpy as np

a = np.int64(1) > np.int64(2)
b = np.float64(2.0) + 1.0
c = 1.0 + np.float64(2.0)
```

```
error: Operator ">" not supported for types "int64" and "int64" (reportGeneralTypeIssues)
error: Operator "+" not supported for types "float64" and "float" (reportGeneralTypeIssues)
error: Operator "+" not supported for types "float" and "float64" (reportGeneralTypeIssues)
```

**VS Code extension or command-line**

Commandline 1.1336, windows 11, python 3.11, numpy 1.26.2

**Comment by erictraut** [comment]
Thanks for the bug report. This regression was introduced in a fix for a previous bug you reported (#6387). As with that one, I'm surprised this one was not covered by an existing test case or caught by the "mypy_primer" regression test suite that we run on ever commit.

This case involves an operator override magic method (e.g. `__lt__` or `__add__`) that is a variable whose type is a callback protocol (an object with a `__call__` method). This is used in numpy's core `number` class:

```python
class number(generic, Generic[_NBit1]):  # type: ignore
    ...
    __lt__: _ComparisonOp[_NumberLike_co, _ArrayLikeNumber_co]
    __le__: _ComparisonOp[_NumberLike_co, _ArrayLikeNumber_co]
    ...
```

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.337, which I just published. It will also be included in a future release of pylance.
